### DISCUSSION TITLE
If Triggered...run shortcut

### DISCUSSION THREAD

**Discussion by sterumbelow** [homebridge-simplisafe3/homebridge-simplisafe3#186] [General]
Hi,

Just a quick question. 

Ive set up an automation to run if the alarm is triggered and want to make sure the triggered state is activated only when the alarm is going off and not when a smoke/leak sensor detects something. 

The reason for doing this is because the SimpliSafe base station on its own isn't very loud. Instead, the automation will trigger HomePod's to play an alarm and also lights to turn on off etc. 

Please can you confirm if the 'Triggered' state means the alarm is already going off and was triggered by door/motion sensors and not by leak/smoke detection.

I want to make sure that 'Triggered' doesn't apply to when a smoke detector or leak sensor goes off!

![image](https://user-images.githubusercontent.com/68124742/108246587-83a34a00-7149-11eb-8762-8c14a6474efb.jpeg)

Thank you,
Ste.

**Comment by shamoon** [comment]
Hi @sterumbelow , thats a great question but I dont actually know the answer. I dont have smoke or water sensors to test. The API, however, does not provide different types of "triggered" for the alarm so I guess if those sensors "trigger" an alarm it would be the same one and they wouldn't be differentiated. But again, Im not sure how those sensors work in terms of triggering the main alarm.

Have you tested the automation at all? (SS has a test mode)

**Reply by sterumbelow** [reply]
I tested the smoke detector yesterday evening and it put out a standard fire alarm noise through the base station and immediately contacted the monitoring company. 

I have no idea whether that is the same state of trigger as an entry sensor going off. 

As for the leak sensor, I know that also functions in a similar way. The base station plays an audible noise, contacts the monitoring centre etc. 

On their website they say consistently refer to the sensors or alarm triggering. That wording leads to me believe that any device going off will put the system in a triggered state, but I don't know for sure. 

I'll have to set my automation to first check each sensor to try and figure out if it's smoke, water or a burglar.

**Reply by sterumbelow** [reply]
As a side note, if you do test any sensors (smoke alarm, water sensor), you should let the monitoring centre know. 

I hit the test button on the smoke detector last night and immediately got a call from the monitoring company. They told me I need to call them to tell them if I'm testing, because a triggered alarm will take priority. 

Wasn't aware of that...

Although Simplisafe does have an actual test mode, which presumably stops the monitoring company being contacted. 😁

**Comment by sterumbelow** [comment]
So, I've set my automation to do this:

If Simplisafe triggers
Check if smoke is detected
• If yes, play smoke alarm sound through HomePod
• If no, then
Check if water is detected
• If yes, play water noise through HomePod
• If no, then
Run automation for burglar entry
• Play burglar alarm noise at full volume
• Turn on all lights and turn them off every two seconds, four times, to simulate flashing lights briefly.
After a set period of time, check whether the alarm is still triggered.
• If it is no longer triggered, stop.
• If still triggered, continue playing burglar alarm sound for a set period
Turn off alarm sound and lights

I think that covers you for each eventuality, unless you're unlucky enough to have a fire, leak, or burglar at the same time.

**Reply by shamoon** [reply]
Yea cool. Since this involves the safety of you and your family please check it, would definitely be curious to see how it works. My biggest concern is that when the alarm triggers and the automation runs the sensor hasn't updated yet in HomeKit and then you might get the wrong "type" of alarm. We have no way to control when the SS API will push these events (e.g. might be alarm first then smoke, or whatever). I guess you'd still get a noisy alarm...

And yes, test mode! Though they'll still call just maybe expecting you to say "All good, just testing"

**Reply by sterumbelow** [reply]
That's very true. 

If a smoke/water sensor had triggered the alarm, I'd expect that its state would've already been updated in Homekit, so running the right part of the automation shouldn't be a problem. 

I deliberately set each part of the automation. You don't need a smoke alarm to be going off for more than 4-5 minutes or you're in major trouble. We also have to remember this automation is just a backup, or additional to the house smoke detectors which would be going off too.

I wonder how long a triggered state lasts though. So when my automation checks again after the set period of time, will me disabling the alarm (in event of it not being a burglar), have changed the state back out of triggered...who knows!

It's worth remembering, as I mentioned above, that all of this is additional to the safety devices in my home. The Simplisafe is unaffected by this and smoke detectors will go off as normal. The worst that can happen is my automation is incorrect and sets off the burglar alarm noise and flashes lights at me instead of playing the smoke alarm noise. 😂

**Reply by shamoon** [reply]
Yea hopefully all that noise and light is enough ht awake you up no matter what. Feel free to report back if you ever discover more details, Im sure others will find it useful!
